### ISSUE TITLE
Read_ply_points_with_properties() with Exact Kernel

### ISSUE THREAD

**Issue by danston** [CGAL/cgal#3696]
## Issue Details

When I try to read points with normals from .ply using CGAL::Exact_predicates_exact_constructions_kernel and the function CGAL::read_ply_points_with_properties(), I get an assertion (reader != NULL). Changing exact kernel to inexact produces the correct result.

### example:
```
#include <fstream>
#include <utility>

#include <CGAL/property_map.h>
#include <CGAL/Simple_cartesian.h>
#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
#include <CGAL/IO/read_ply_points.h>

using Kernel = CGAL::Exact_predicates_exact_constructions_kernel;

using Point_3 = typename Kernel::Point_3;
using Vector_3 = typename Kernel::Vector_3;

using Point_with_normal = std::pair<Point_3, Vector_3>;

using Point_map = CGAL::First_of_pair_property_map<Point_with_normal>;
using Normal_map = CGAL::Second_of_pair_property_map<Point_with_normal>;

int main(int argc, char **argv) {

  const auto file_name = argv[1];
  std::vector<Point_with_normal> data;
  std::ifstream in(file_name);

  if (
    !in ||
    !CGAL::read_ply_points_with_properties(in,
    std::back_inserter(data),
    CGAL::make_ply_point_reader(Point_map()),
    CGAL::make_ply_normal_reader(Normal_map()))) {

    std::cerr << "Error: cannot read file " << file_name << std::endl;
    return EXIT_FAILURE;
  }

  const Point_3& p = data[0].first;
  const Vector_3& n = data[0].second;
    
  std::cout.precision(15);
  std::cout << "Point (" << p << ") with normal (" << n << ")" << std::endl;

  return EXIT_SUCCESS;
}
```

### cmake:
```
project(test)

cmake_minimum_required(VERSION 2.8.10)
set(CMAKE_CXX_STANDARD 11)

find_package(CGAL QUIET COMPONENTS Core)
if(CGAL_FOUND)

  include(${CGAL_USE_FILE})
  include(CGAL_CreateSingleSourceCGALProgram)

  create_single_source_cgal_program(
    "test.cpp")

else()
  message(WARNING 
  "This program requires the CGAL library, and will not be compiled.")
endif()
```

### ply:
```
ply
format ascii 1.0
comment Generated by the CGAL library
comment label -1 unclassified
comment label 0 ground
comment label 1 facade
comment label 2 roof
comment label 3 vegetation
element vertex 3
property double x
property double y
property double z
property double nx
property double ny
property double nz
property uchar red
property uchar green
property uchar blue
property int training
property int label
end_header
-195.30807495117190342 -82.15155029296875 46.991355895996086645 -0.80236899852752685547 -0.57400602102279663086 0.16346600651741030608 0 255 0 3 3
-271.105072021484375 -44.64154434204102273 43.307109832763671875 -0.3638510107994080145 -0.9114519953727722168 0.19201199710369110107 0 255 0 -1 3
-296.8485107421875 -13.800042152404790485 49.309192657470703125 0.0046720001846551903815 0.61861199140548706055 0.78568297624588012695 0 255 0 -1 3
```

### error:
```
CGAL error: assertion violation!
Expression : reader != NULL
File       : /usr/local/include/CGAL/IO/read_ply_points.h
Line       : 520
Explanation: 
Refer to the bug-reporting instructions at https://www.cgal.org/bug_report.html
libc++abi.dylib: terminating with uncaught exception of type CGAL::Assertion_exception: CGAL ERROR: assertion violation!
Expr: reader != NULL
File: /usr/local/include/CGAL/IO/read_ply_points.h
Line: 520
Abort trap: 6
```

## Environment

* Operating system: Mac and Windows
* Compiler: clang and visual studio
* Release or debug mode: both
* CGAL version: 4.13
* Boost version: 1.68

**Comment by sgiraudot** [comment]
Thanks for the report!

This should be fixed by https://github.com/CGAL/cgal/pull/3703.

**Comment by danston** [comment]
Great. Thank you!

**Comment by lrineau** [comment]
Fixed by the merge of #3703.
